### PR TITLE
Waypoint improvements

### PR DESCRIPTION
**Pull request by yhoazk**
Removed dead code and fixed the issue of the car staying in the middle of the intersection.

### CODE REVIEW

**Review by dabavil — APPROVED** [review]

File: ros/src/tl_detector/tl_detector.py [hunk 1]
```diff
@@ -265,6 +281,8 @@ def traffic_cb(self, msg):
         Finds next closest traffic light in front, takes its ground truth state
         and publishes to /traffic_waypoint
         """
+        self.tf_tst = msg.lights[0].state
+        return
```
Comment (avrabe) [inline ros/src/tl_detector/tl_detector.py]: Is this return here right?

**Review by avrabe — CHANGES_REQUESTED** [review]

File: ros/src/tl_detector/tl_detector.py (hunk 1 above)
Comment (vguerra) [reply ros/src/tl_detector/tl_detector.py]: i think it wont hurt to have a return here but could be removed as it brings no value.

File: ros/src/tl_detector/tl_detector.py [hunk 2]
```diff
@@ -219,17 +220,37 @@ def get_next_tl_waypoint_index(self, stop_line_positions):
...
+            light = True
+        else:
+            self.in_range = False
+
+        if self.in_range and not self.last_in_range:
+            rospy.logwarn("tl_detector: TL in range SL_WP: {}, Car_WP: {}".format(tl_wp_idx, car_wp_idx))
+            self.last_in_range = self.in_range
+        
+        return tl_wp_idx
+        """
```
Comment (vguerra) [inline ros/src/tl_detector/tl_detector.py]: i would prefer to remove the commeted code here, you can always check prev versions in git.

**Review by vguerra — CHANGES_REQUESTED** [review]

**Review by avrabe — APPROVED** [review]

File: ros/src/waypoint_updater/waypoint_updater.py [hunk 3]
```diff
@@ -6,7 +6,7 @@
 from geometry_msgs.msg import PoseStamped
 from geometry_msgs.msg import TwistStamped
 from styx_msgs.msg import Lane, Waypoint
-from styx_msgs.msg import TrafficLightArray
+from styx_msgs.msg import TrafficLightArray, TrafficLight
```
Comment (asimonov) [inline ros/src/waypoint_updater/waypoint_updater.py:9]: I am not following, but are we actually using TrafficLight/Array here? I think we just receive index of next traffic light waypoint?

File: ros/src/tl_detector/tl_detector.py [hunk 4]
```diff
@@ -219,33 +212,30 @@ def get_next_tl_waypoint_index(self, stop_line_positions):
...
-                    tl_wp_idx = i
-                    min_dist = dist
 
+        light_dist = [0 < (self.car_direction *(line_index - car_wp_idx)) < INDEX_DISTANCE_THRESHOLD
+                        for line_index in self.stop_lines_wp_idxs]
+        
         self.last_in_range = self.in_range
-        if min_dist < 1e+10:
+
+        if any(light_dist):
```
Comment (asimonov) [inline ros/src/tl_detector/tl_detector.py:230]: thanks :) I know 1e+10 looked ugly but was in a rush to finish optimising on saturday

**Review by asimonov — APPROVED** [review]
thank you!

**Comment by asimonov** [comment]
Sorry, I was not following for couple of days. I am not sure if Victor's changes were incorporated. I assume they were and going to merge

**Comment by asimonov** [comment]
I hope I resolved it correctly

**Comment by asimonov** [comment]
@avrabe Ralf, For Travis build I see this line in the log:
https://travis-ci.org/Kairos-Automotive/carla-brain/builds/295534971#L803
but overall the state is reported as 'All Checks Have Passed'.
Do you think it is a bug?

**Pull request merged**